src/Sylius/Component/Core/spec/Promotion/Action/ItemFixedDiscountActionSpec.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Action;
+
+use Doctrine\Common\Collections\Collection;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Model\PromotionInterface;
+use Sylius\Component\Core\Promotion\Action\ItemDiscountAction;
+use Sylius\Component\Core\Promotion\Filter\FilterInterface;
+use Sylius\Component\Originator\Originator\OriginatorInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ItemFixedDiscountActionSpec extends ObjectBehavior
+{
+    function let(
+        FactoryInterface $adjustmentFactory,
+        OriginatorInterface $originator,
+        FilterInterface $priceRangeFilter,
+        FilterInterface $taxonFilter
+    ) {
+        $this->beConstructedWith($adjustmentFactory, $originator, $priceRangeFilter, $taxonFilter);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Action\ItemFixedDiscountAction');
+    }
+
+    function it_is_discount_action()
+    {
+        $this->shouldHaveType(ItemDiscountAction::class);
+    }
+
+    function it_applies_percentage_discount_on_every_unit_in_order(
+        $adjustmentFactory,
+        $originator,
+        $priceRangeFilter,
+        $taxonFilter,
+        AdjustmentInterface $promotionAdjustment1,
+        AdjustmentInterface $promotionAdjustment2,
+        Collection $originalItems,
+        Collection $units,
+        OrderInterface $order,
+        OrderItemInterface $orderItem1,
+        OrderItemInterface $orderItem2,
+        OrderItemInterface $orderItem3,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion
+    ) {
+        $order->getItems()->willReturn($originalItems);
+        $originalItems->toArray()->willReturn([$orderItem1, $orderItem2, $orderItem3]);
+
+        $priceRangeFilter
+            ->filter([$orderItem1, $orderItem2, $orderItem3], ['amount' => 500, 'filters' => ['taxons' => ['testTaxon']]])
+            ->willReturn([$orderItem1, $orderItem2])
+        ;
+        $taxonFilter
+            ->filter([$orderItem1, $orderItem2], ['amount' => 500, 'filters' => ['taxons' => ['testTaxon']]])
```

Comment:
Try to use real examples in spec also:
`testTaxon` -> `T-Shirts`
`Test description` -> `Christmas sale`
etc.